packages/zipkin-encoder-thrift/src/index.js

```diff
@@ -0,0 +1,113 @@
+const {option: {fromNullable}} = require('zipkin');
+const thriftTypes = require('./gen-nodejs/zipkinCore_types');
+const {TBufferedTransport, TBinaryProtocol} = require('thrift');
+
+function toThriftEndpoint(ep, serviceName) {
+  return new thriftTypes.Endpoint({
+    service_name: serviceName.getOrElse('unknown'),
+    ipv4: ep.host,
+    port: ep.port
+  });
+};
+
+function toThriftAnnotation(ann, thriftEndpoint) {
+  const res = new thriftTypes.Annotation({
+    timestamp: ann.timestamp, // must be in micros
+    value: ann.value,
+  });
+  if (thriftEndpoint) {
+    res.host = thriftEndpoint;
+  }
+  return res;
+};
+
+function toThriftBinaryAnnotation(key, value, thriftEndpoint) {
+  const res = new thriftTypes.BinaryAnnotation({
+    key: key,
+    value: value,
+    annotation_type: thriftTypes.AnnotationType.STRING
+  });
+  if (thriftEndpoint) {
+    res.host = thriftEndpoint;
+  }
+  return res;
+};
+
+function toThriftAddress(key, endpoint) {
+  const value = Buffer.from([1]);
+  const res = new thriftTypes.BinaryAnnotation({
+    key: key,
+    value,
+    annotation_type: thriftTypes.AnnotationType.BOOL,
+    host: toThriftEndpoint(endpoint, fromNullable(endpoint.serviceName))
+  });
+  return res;
+};
+
+function toThriftSpan(span) {
+  let startedSpan = true;
+  const res = new thriftTypes.Span();
+
+  res.id = span.traceId.spanId;
+  span.traceId._parentId.ifPresent(id => {
+    res.parent_id = id;
+  });
+
+  const traceId = span.traceId.traceId;
+  if (traceId.length <= 16) {
+    res.trace_id = traceId;
+  } else {
+    res.trace_id_high = traceId.substr(0, 16);
+    res.trace_id = traceId.substr(traceId.length - 16);
+  }
+  res.name = span.name.getOrElse('unknown');
+  const thriftEndpoint = toThriftEndpoint(span.endpoint, span.service);
+
+  let addressKey = 'sa';
+  res.annotations = span.annotations.map(ann => {
+    // In the RPC span model, the client owns the timestamp and duration of the
```

Comment:
after this change... not too long from now, I'll clean up this which is repeated and has been replaced with "span.shared" present in span v2